### ISSUE TITLE
EFS CSI Driver pod remain in pending for Fargate node due to Tolerations

### ISSUE THREAD

**Issue by parikshitataws** [kubernetes-sigs/aws-efs-csi-driver#340]
/kind bug

**What happened?**
EFS CSI Driver pod remain in pending for Fargate node due to Tolerations (Tolerations:     op=Exists)

**What you expected to happen?**
EFS CSI Driver pod should not try to schedule on Fargate node 

**How to reproduce it (as minimally and precisely as possible)?**
Create a Fargate profile and run at least one pod in Fargate. This is cause a Fargate node to show up in the cluster 
```
kg node
NAME                                                         STATUS   ROLES    AGE     VERSION
fargate-ip-192-168-188-106.ap-southeast-2.compute.internal   Ready    <none>   2d23h   v1.18.8-eks-7c9bda
ip-192-168-40-122.ap-southeast-2.compute.internal            Ready    <none>   34d     v1.18.9-eks-d1db3c
ip-192-168-92-102.ap-southeast-2.compute.internal            Ready    <none>   2d20h   v1.18.9-eks-d1db3c
```

Apply EFS CSI Driver : 
kubectl apply -k "github.com/kubernetes-sigs/aws-efs-csi-driver/deploy/kubernetes/overlays/stable/ecr/?ref=release-1.0"

List the pods 
```
ksg pod -o wide
NAME                                 READY   STATUS    RESTARTS   AGE     IP               NODE                                                NOMINATED NODE   READINESS GATES
aws-node-gpwfp                       1/1     Running   0          34d     192.168.40.122   ip-192-168-40-122.ap-southeast-2.compute.internal   <none>           <none>
aws-node-vq7s6                       1/1     Running   0          2d20h   192.168.92.102   ip-192-168-92-102.ap-southeast-2.compute.internal   <none>           <none>
cluster-autoscaler-58f66bff6-xrddm   1/1     Running   0          31d     192.168.38.153   ip-192-168-40-122.ap-southeast-2.compute.internal   <none>           <none>
coredns-67997b9dbd-2dn97             1/1     Running   0          2d19h   192.168.83.14    ip-192-168-92-102.ap-southeast-2.compute.internal   <none>           <none>
coredns-67997b9dbd-wtd9t             1/1     Running   0          31d     192.168.42.123   ip-192-168-40-122.ap-southeast-2.compute.internal   <none>           <none>
efs-csi-node-4ftt4                   0/3     Pending   0          15s     <none>           <none>                                              <none>           <none>
efs-csi-node-bdhjb                   3/3     Running   0          5m17s   192.168.40.122   ip-192-168-40-122.ap-southeast-2.compute.internal   <none>           <none>
efs-csi-node-qc46n                   3/3     Running   0          5m21s   192.168.92.102   ip-192-168-92-102.ap-southeast-2.compute.internal   <none>           <none>
kube-proxy-kx4p6                     1/1     Running   0          34d     192.168.40.122   ip-192-168-40-122.ap-southeast-2.compute.internal   <none>           <none>
kube-proxy-wzrbf                     1/1     Running   0          2d20h   192.168.92.102   ip-192-168-92-102.ap-southeast-2.compute.internal   <none>           <none>
```

```
ksd pod efs-csi-node-4ftt4
Name:                 efs-csi-node-4ftt4
Namespace:            kube-system
Priority:             2000001000
Priority Class Name:  system-node-critical
Node:                 <none>
Labels:               app=efs-csi-node
                      controller-revision-hash=5f49f95c56
                      pod-template-generation=3
Annotations:          kubernetes.io/psp: eks.privileged
Status:               Pending
IP:
IPs:                  <none>
Controlled By:        DaemonSet/efs-csi-node
Containers:
  efs-plugin:
    Image:      602401143452.dkr.ecr.us-west-2.amazonaws.com/eks/aws-efs-csi-driver:v1.0.0
    Port:       9809/TCP
    Host Port:  9809/TCP
    Args:
      --endpoint=$(CSI_ENDPOINT)
      --logtostderr
      --v=5
    Liveness:  http-get http://:healthz/healthz delay=10s timeout=3s period=2s #success=1 #failure=5
    Environment:
      CSI_ENDPOINT:  unix:/csi/csi.sock
    Mounts:
      /csi from plugin-dir (rw)
      /etc/amazon/efs from efs-utils-config (rw)
      /var/lib/kubelet from kubelet-dir (rw)
      /var/run/efs from efs-state-dir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-tzm8b (ro)
  csi-driver-registrar:
    Image:      602401143452.dkr.ecr.us-west-2.amazonaws.com/eks/csi-node-driver-registrar:v1.3.0
    Port:       <none>
    Host Port:  <none>
    Args:
      --csi-address=$(ADDRESS)
      --kubelet-registration-path=$(DRIVER_REG_SOCK_PATH)
      --v=5
    Environment:
      ADDRESS:               /csi/csi.sock
      DRIVER_REG_SOCK_PATH:  /var/lib/kubelet/plugins/efs.csi.aws.com/csi.sock
      KUBE_NODE_NAME:         (v1:spec.nodeName)
    Mounts:
      /csi from plugin-dir (rw)
      /registration from registration-dir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-tzm8b (ro)
  liveness-probe:
    Image:      602401143452.dkr.ecr.us-west-2.amazonaws.com/eks/livenessprobe:v2.0.0
    Port:       <none>
    Host Port:  <none>
    Args:
      --csi-address=/csi/csi.sock
      --health-port=9809
    Environment:  <none>
    Mounts:
      /csi from plugin-dir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-tzm8b (ro)
Conditions:
  Type           Status
  PodScheduled   False
Volumes:
  kubelet-dir:
    Type:          HostPath (bare host directory volume)
    Path:          /var/lib/kubelet
    HostPathType:  Directory
  registration-dir:
    Type:          HostPath (bare host directory volume)
    Path:          /var/lib/kubelet/plugins_registry/
    HostPathType:  Directory
  plugin-dir:
    Type:          HostPath (bare host directory volume)
    Path:          /var/lib/kubelet/plugins/efs.csi.aws.com/
    HostPathType:  DirectoryOrCreate
  efs-state-dir:
    Type:          HostPath (bare host directory volume)
    Path:          /var/run/efs
    HostPathType:  DirectoryOrCreate
  efs-utils-config:
    Type:          HostPath (bare host directory volume)
    Path:          /etc/amazon/efs
    HostPathType:  DirectoryOrCreate
  default-token-tzm8b:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-tzm8b
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  kubernetes.io/arch=amd64
                 kubernetes.io/os=linux
Tolerations:     op=Exists
                 node.kubernetes.io/disk-pressure:NoSchedule op=Exists
                 node.kubernetes.io/memory-pressure:NoSchedule op=Exists
                 node.kubernetes.io/network-unavailable:NoSchedule op=Exists
                 node.kubernetes.io/not-ready:NoExecute op=Exists
                 node.kubernetes.io/pid-pressure:NoSchedule op=Exists
                 node.kubernetes.io/unreachable:NoExecute op=Exists
                 node.kubernetes.io/unschedulable:NoSchedule op=Exists
Events:
  Type     Reason             Age                From                Message
  ----     ------             ----               ----                -------
  Warning  FailedScheduling   44s (x2 over 45s)  default-scheduler   0/3 nodes are available: 1 Too many pods, 2 node(s) didn't have free ports for the requested pod ports.
  Normal   NotTriggerScaleUp  41s                cluster-autoscaler  pod didn't trigger scale-up (it wouldn't fit if a new node is added): 1 node(s) didn't have free ports for the requested pod ports
```

Now edit DaemonSet (efs-csi-node) and remove tolerations to allow it to schedule only on EC2 worker nodes. 

**Anything else we need to know?**:

**Environment**
- Kubernetes version (use `kubectl version`):
k version --short
Client Version: v1.20.3
Server Version: v1.18.9-eks-d1db3c

- Driver version:

**Comment by brankom-daon** [comment]
Same here... 
`NAME                                         READY   STATUS    RESTARTS   AGE
aws-node-jfw8j                               1/1     Running   0          6h3m
coredns-7f85bf9964-s259n                     1/1     Running   0          6h23m
coredns-7f85bf9964-xvmwt                     1/1     Running   0          6h23m
efs-csi-node-wd965                           0/3     Pending   0          17m
efs-csi-node-xxp55                           3/3     Running   0          5h34m
kube-proxy-mwrjw                             1/1     Running   0          6h3m
metrics-server-76f8d9fc69-dt9z8              1/1     Running   0          5h52m
sealed-secrets-controller-7c766b885b-rh2k4   1/1     Running   0          5h40m
`

`Events:
  Type     Reason            Age                  From               Message
  ----     ------            ----                 ----               -------
  Warning  FailedScheduling  29s (x5 over 3m12s)  default-scheduler  0/2 nodes are available: 1 Too many pods, 1 node(s) didn't have free ports for the requested pod ports.
`

**Comment by brankom-daon** [comment]
Having exact same thing here

**Comment by wongma7** [comment]
issue to reduce scope of default tolerations: https://github.com/kubernetes-sigs/aws-efs-csi-driver/issues/363

fix to limit the pod to only ec2 instances using antiaffinity (will be released w/ 1.2): https://github.com/kubernetes-sigs/aws-efs-csi-driver/pull/329
